ts/message/index.ts

```diff
@@ -0,0 +1,67 @@
+import rpcProtoStr from './rpc.proto'
+import protons = require('protons')
+
+const rpcProto = protons(rpcProtoStr)
+
+export interface SubOpts {
+  subscribe?: boolean
+  topicID?: string
+}
+
+export interface Message {
+  from?: Buffer
+  data?: Buffer
+  seqno?: Buffer
+  topicIDs: string[]
+  signature?: Buffer
+  key?: Buffer
+}
+
+/**
+ * Same as Message, but `from` is an optional string
+ */
+export interface InMessage {
```

Comment:
https://stackoverflow.com/questions/43159887/make-a-single-property-optional-in-typescript to avoid duplication